services/git-service-impl/src/main/java/io/fabric8/launcher/service/gitlab/impl/GitLabServiceImpl.java

```diff
@@ -146,9 +146,12 @@ public void deleteRepository(String repositoryName) throws IllegalArgumentExcept
     }
 
     @Override
-    public GitHook createHook(GitRepository repository, URL webhookUrl, String... events) throws IllegalArgumentException {
+    public GitHook createHook(GitRepository repository, String secret, URL webhookUrl, String... events) throws IllegalArgumentException {
```

Comment:
what invalid characters that would be? A secret could have anything afaik :)

Reply:
Just giving it the ol' naked eye glace; looked like it was assembling a query string in the GitLab impl and the space was unescaped, which would break the query string. :)  If no problem: no problem!

Reply:
Good catch, yeah, that could definitely be an issue. We need to escape that

Reply:
Or at least have a test for that scenario

Reply:
Good catch indeed !! I love open source development :)